src/modules/bbcore.videoRecorder.js

```diff
@@ -109,18 +131,42 @@ BBCore.prototype.startVideoRecorder = function (opts, recordComplete) {
     var inst = this;
     // get recorder and inject into target
     this.getVideoRecorder(rec_opts, function (data) {
-        if (!inst.currentVideoId && data.info.vid_id) {
-            inst.currentVideoId = data.info.vid_id;
-        }
+        inst.currentVideoId = data.info.vid_id;
+
         console.log('startVideoRecorder :' + inst.currentVideoId);
-        inst.__vidRecHndl.html(data.info.content);
+        // `getVideoRecorder` returns html and js in the form of a string. In the old version of BBCore,
+        // script tags were handled by jquery (by use of the eval method). Since we don't want to rely
+        // on jQuery or end up having double event listeners, we go with the jquery way of mounting this
+        // js if the cleint already supplies jQuery, or hardcode it ourselves in jQuery isn't around.
+        if (window.jQuery) {
+            this.__vidRecHndl = opts.target ? jQuery(opts.target) : jQuery('body').append('<div id="b2recorder"></div>');
+            inst.__vidRecHndl.html(data.info.content);
+        } else {
+          window.addEventListener('message', function(event) {
+              if (~event.origin.indexOf('app.bombbomb.com')) {
```

Comment:
What's the `~` operator?

Reply:
bitwise not. Had to look it up lol. Pretty much it adds 1 and then `!`s it. I typically would never actually use this code but its exactly what comes from `data.info.content`. You can read more about it here: https://wsvincent.com/javascript-tilde/#:~:text=In%20JavaScript%2C%20the%20tilde%20~%20Bitwise,a%20String%20object%20passed%20in.&text=So%20if%20%2D1%20is%20returned,into%200%20which%20is%20falsy.

Reply:
Aha, thanks for the link! Interesting bit operating...